src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/ServiceProviderContainerTests.cs

```diff
@@ -156,6 +156,53 @@ public void ResolvesServiceMixedServiceAndOptionalStructConstructorArgumentsReli
             }
         }
 
+        public enum TheEnum
+        {
+            HelloWorld = -1,
+            NiceWorld = 0,
+            GoodByeWorld = 1,
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void ResolvesConstantValueTypeServicesCorrectly(bool useScoped)
+        {
+            var serviceCollection = new ServiceCollection();
+            if (useScoped)
```

Comment:
No transients?

Reply:
Should we test struct services that depende on other services?

Reply:
> No transients?

Added.

> Should we test struct services that depende on other services?

We have this test which is currently disabled:

https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/ServiceProviderContainerTests.cs#L211-L225

I opened https://github.com/dotnet/runtime/issues/42160 to fix this scenario.

Reply:
Do we have:

- `IEnumerable<Struct>` tests
- `IEnumerable<InterfaceWithStructProperty>` tests
- `InterfaceWithStructProperty` test?

Reply:
I have IEnumerable(Struct) tests in this PR.

I’m not sure what you mean by InterfaceWithStructProperty tests. Like a class that implements an interface, and it has an int property on the interface?

Reply:
> Like a class that implements an interface, and it has an int property on the interface? 

Yep